doc/en/aseba-language.rst

```diff
@@ -0,0 +1,637 @@
+
+    state = 0
+    call leds.bottom.left(0,0,32)
+    call leds.bottom.right(0,32,0)
+    call leds.top(32,0,0)
+
+While the initialization of ``state`` could have been done in its
+declaration, the initialization of the leds must be done by statements.
+When programming a robot, you will usually want to define some event
+that will re-initialize the state of the robot. This is possible by
+writing the statements within a subroutine and calling it from the event
+handler. It is also possible to call the subroutine as part of the
+program initialization even though it has not yet been declared:
+
+::
+
+    var state
+
+    callsub init  # Initialize the program
+
+    # Subroutine for initialization
+    sub init
+        state = 0
+        call leds.bottom.left(0,0,32)
+        call leds.bottom.right(0,32,0)
+        call leds.top(32,0,0)
+
+    # Re-initialize when center button is touched
+    onevent button.center
+        callsub init
+
+Sending external events
+-----------------------
+
+The programme can send external events by using the ``emit`` keyword,
+followed by the name of the event and the name of the variable to send,
+if any. If a variable is provided, the size of the event must match the
+size of the
+`argument <http://en.wikipedia.org/wiki/Argument_%28computer_science%29>`__
+to be emitted. Instead of a variable, array constructors and
+mathematical expressions can also be used in more complex situations.
+Events allow the programme to trigger the execution of code on another
+node or to communicate with an external programme.
+
+::
+
+    onevent ir_sensors
+    emit sensors_values proximity_sensors_values
+
+Native functions
+----------------
+
+We designed the Aseba language to be simple in order to allow a quick
+understanding even by novice developers and to implement the virtual
+machine efficiently on a micro-controller. To perform complex or
+resource-intensive processing, we provide native functions that are
+implemented in native code for efficiency. For instance, a native
+function is the natural way to implement a scalar product.
+
+Native functions are safe, as they specify and check the size of their
+arguments, which can be constants, variables, array accesses, array
+constructors and expressions. In the case of an array, you can access
+the whole array, a single element, or a sub-range of the array. Native
+functions take their arguments by
+`reference <http://en.wikipedia.org/wiki/Call_by_reference#Call_by_reference>`__
+and can modify their contents but do not return any value. You can use
+native functions through the ``call`` keyword.
+
+Example:
+
+::
+
+    var a[3] = 1, 2, 3
+    var b[3] = 2, 3, 4
+    var c[5] = 5, 10, 15
+    var d
+    call math.dot(d, a, b, 3)
+    call math.dot(d, a, c[0:2], 3)
+    call math.dot(a[0], c[0:2], 3)
+
+What to read next?
+------------------
+
+You might be interested to read:
+
+-  `Description of the native functions standard library <asebastdnative.rs
```

Comment:
Broken link

Reply:
Oh, yeah, I meant to either fix that - but that one of the link I considered for removal. Do you think it's needed ? Maybe we neeed to setup read the doc before deciding

Reply:
I agree, we should setup read the doc first.